src/app/core/interceptors/loader-interceptor/loader.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from "@angular/core";
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
+import { Observable } from "rxjs";
+
+import { LoadingService } from "../../services/index";
+import { debounceTime, finalize } from "rxjs/operators";
+
+
+@Injectable()
+export class LoaderInterceptor implements HttpInterceptor {
+
+    private activeRequests: number = 0;
+    private loadingService: LoadingService;
+
+    constructor(loadingService: LoadingService) {
+        this.loadingService = loadingService;
+    }
+
+    public intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        if (this.activeRequests === 0) {
+            this.loadingService.startLoading();
+        }
+
+        this.activeRequests++;
+        return next.handle(request).pipe(
+            debounceTime(5000),

```

Comment:
I don't think it's a good idea to delay fetched data to user for 5 seconds

Reply:
i forgot to remove it ) it was just for purpose of experimenting